lib/tapioca/commands/dsl_generate.rb

```diff
@@ -32,6 +32,8 @@ def execute
 
         say("All operations performed in working directory.", [:green, :bold])
         say("Please review changes and commit them.", [:green, :bold])
+      ensure
```

Comment:
It's surprising that we're trying to create the file even if something went wrong earlier.

Reply:
Since `gitattribute` file creation makes sure that the output folder exists before creating a file, I figured it would be good to unconditionally generate a gitattribute file in case any RBI files have already been generated by the time we hit an error.

Reply:
If something went wrong earlier, we do not have any guaranty that `@outpath` even exist.

Also I'm wondering if the `.gitattribute` should be moved to `tapioca init` rather than rewriting it every time we generate the RBIs?

Reply:
> If something went wrong earlier, we do not have any guaranty that @outpath even exist.

That was exactly my point above. The way [`GitAttributes.create_gitattributes_file` method](https://github.com/Shopify/tapioca/pull/1604/files#diff-d9d26e65534c05831a2c22654a3e66df0b3cf6df8650068833598b08eaa8cae8R26-R29) works is to first check if the folder exists or not. If the folder does not exist, it does not create any file. So, this is the safest way to generate `.gitattributes` file for any RBI that may be generated in the output directory.

> Also I'm wondering if the .gitattribute should be moved to tapioca init rather than rewriting it every time we generate the RBIs?

I don't think that would be a great thing to add. People could delete their `rbi` folder, or do a bunch of other crazy things without needing to run `init` command again. A `gem` or `dsl` command should make sure a valid `.gitattributes` file is present in the output path if any RBI files are generated.

This would also make sure that people who upgrade their Tapioca version and generate new gem/dsl RBIs will get the `.gitattributes` file as well.